compiler/rustc_span/src/lib.rs

```diff
@@ -378,24 +378,38 @@ pub struct SpanData {
     /// Information about where the macro came from, if this piece of
     /// code was created by a macro expansion.
     pub ctxt: SyntaxContext,
+    pub parent: Option<LocalDefId>,
```

Comment:
Do I understand correctly that `parent: None` is equivalent to using the whole crate (`LocalDefId=0`)  and its span (`0 .. N`) as a parent? Perhaps it's better to avoid the `Option` then?

Reply:
Having `None` means we do not try encoding the span relative to a parent. Using `Some(CRATE_DEF_ID)` would be trying to encode relative to the crate span, ie. `lib.rs`'s span.

Reply:
I mean the crate span starts from `BytePos == 0`, so absolute spans and spans relative to the crate span are the same thing.

Reply:
For absolute spans, we do not hash their byte position. Instead, we hash the file name and the position within that file.
Conflating both cases would mean hashing absolute spans as beyond-the-end offsets in lib.rs.